wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreatePasswordSecuredLink/CreateSecureGuestLinkViewController.swift

```diff
@@ -0,0 +1,130 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Down
+import UIKit
+import WireCommonComponents
+
+class CreateSecureGuestLinkViewController: UIViewController, CreatePasswordSecuredLinkViewModelDelegate {
+
+    // MARK: - Properties
+
+    typealias ViewColors = SemanticColors.View
+    typealias LabelColors = SemanticColors.Label
+    typealias SecuredGuestLinkWithPasswordLocale = L10n.Localizable.SecuredGuestLinkWithPassword
+
+    private lazy var generatePasswordButton: SecondaryTextButton = {
+        let button = SecondaryTextButton(
+            fontSpec: FontSpec.buttonSmallSemibold,
+            insets: UIEdgeInsets(top: 8, left: 8, bottom: 8, right: 8)
+        )
+
+        button.setTitle(SecuredGuestLinkWithPasswordLocale.GeneratePasswordButton.title, for: .normal)
+        button.setImage(Asset.Images.shield.image, for: .normal)
+        button.addTarget(self, action: #selector(generatePasswordButtonTapped), for: .touchUpInside)
+        button.imageEdgeInsets.right = 10.0
+        return button
+    }()
+
+    private var viewModel = CreateSecureGuestLinkViewModel()
+
+    private let warningLabel: UILabel = {
+        var paragraphStyle = NSMutableParagraphStyle()
+        var label = UILabel()
+        label.numberOfLines = 0
+        label.adjustsFontForContentSizeCategory = true
+        label.attributedText = .markdown(from: SecuredGuestLinkWithPasswordLocale.WarningLabel.title, style: .warningLabelStyle)
+        return label
+    }()
+
+    // MARK: - Override methods
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        viewModel.delegate = self
```

Comment:
I would move this to init of viewModel so we don't end up in a case where we don't know why some feedback was not applied.

```
private lazy var viewModel: CreateSecureGuestLinkViewModel {
   return CreateSecureGuestLinkViewModel(delegate: self)
}()
```